### ISSUE TITLE
Current Qpack is a very simple one

### ISSUE THREAD

**Issue by ddragana** [mozilla/neqo#49]
the qpack algorithm needs optimization.

**Comment by martinthomson** [comment]
I wouldn't put this at the top of the list for a while.  Simple will suffice until we have more of the stack in place.  I'm sure that when it becomes a performance issue we'll get around to tweaking it.

**Comment by ddragana** [comment]
Yes, this is for later, way later. The issue is just to know that it exists.

**Comment by larseggert** [comment]
@martinthomson is this OBE?

**Comment by martinthomson** [comment]
We could maybe do better.  Basically, Dragana and I decided that a simple strategy would suffice in the short term, but we knew (from talking to others) that a little bit of work could probably improve performance a fair bit.

There's a lot of performance on the table here, particularly in those first few round trips where the congestion window is small.  If we can pack one or two extra requests in, that could be a big deal for page load metrics.  I'd say keep this (or something with a less casual name) on the books.

**Comment by larseggert** [comment]
Related discussion in https://github.com/mozilla/neqo/pull/2632#discussion_r2089914571
